src/app/application/components/data-source/DataLoader.vue

```diff
@@ -91,7 +91,7 @@ import ErrorBlock from '@/app/common/ErrorBlock.vue'
 import LoadingBlock from '@/app/common/LoadingBlock.vue'
 
 const props = withDefaults(defineProps<{
-  data?: TypeOf<T>[]
+  data?: unknown[]
```

Comment:
Why do we move away from generic to unknown?

Reply:
umm, I can hardly remember 😆 

eeem, I think it was because `:data="[]"` can be a mix/an array of differing types. `:src=""` is the thing we need to infer the type from, whereas we never need to infer the types of `:data="[]"` because we'll already have sourced those from a different DataSource component.

Reply:
should this conversation be persisted as a code comment?